geetools/ee_dictionary.py

```diff
@@ -80,3 +82,42 @@ def getMany(self, list: list | ee.List) -> ee.List:
                 d.geetools.getMany(["foo", "bar"]).getInfo()
         """
         return ee.List(list).map(lambda key: self._obj.get(key))
+
+    def toTable(self, valueType: ee.List | ee.Dictionary | Any = Any) -> ee.FeatureCollection:
+        """Convert a ee.Dictionary to a ee.FeatureCollection with no geometries (table).
+
+        The keys will always be stored in `system:index` column.
+
+        Parameters:
+            valueType: this will define how to process the values. In case of
+            ee.List the values will be stored in columns named `value_<position>`.
+            In case of a ee.Dictionary, column names will be created from the keys.
+            For any other type, it will return a table with one feature with
+            one column per key.
+
+        Returns:
+            a ee.FeatureCollection in which the keys of the ee.Dictionary are
+            in the `system:index` and the values are in new columns.
+        """
+
+        def features_from_dict(key, value) -> ee.Feature:
+            key = ee.String(key)
+            feat = ee.Feature(None, {"system:index": key})
+            return feat.set(ee.Dictionary(value))
+
+        def features_from_list(key, value) -> ee.Feature:
+            key = ee.String(key)
+            feat = ee.Feature(None, {"system:index": key})
+            value = ee.List(value)
+            keys = ee.List.sequence(1, value.size())
+            keys = keys.map(lambda k: ee.String("value_").cat(ee.Number(k).toInt()))
+            properties = ee.Dictionary.fromLists(keys, value)
+            return feat.set(properties)
+
+        if valueType == ee.Dictionary:
+            features = self._obj.map(features_from_dict).values()
+        elif valueType == ee.List:
+            features = self._obj.map(features_from_list).values()
+        else:
+            return ee.FeatureCollection([ee.Feature(None, self._obj)])
```

Comment:
This leads to 2 very differnet ways of treating the dictionnary keys and I think it's more confusing than helping. At the end of the day, the use case is when the keys of the dict are the future `system:index` of the features. The case were the keys are simply treated as column names is so trivial (`f = ee.Feature(, None, self._obj)`) that I think it should not be accepted as an input of this function. 

Narrowing down the use cases will also make it easier to understand the first sentence of the docstring "The keys will always be stored in `system:index` column."